Core.lua

```diff
@@ -589,9 +589,16 @@ do
 	-- @name IsRowVisible
 	-- @description Checks if a row is currently being shown
 	-- @usage st:IsRowVisible(realrow)
-	-- @thanks sapu94
 	local function IsRowVisible(self, realrow)
-		return (realrow > self.offset and realrow <= (self.displayRows + self.offset))
+		-- Scan through all on-screen rows, checking their real data row numbers for a match.
+		local firstVisibleIdx = 1 + (self.offset or 0);
+		local lastVisibleIdx = (firstVisibleIdx + self.displayRows) - 1;
```

Comment:
This line, explained: We need to know the final row index "visible on screen", in the `.filtered` array...

So we know we're showing indices starting from `firstVisibleIdx` (such as `1`), and showing `self.displayRows` amount of rows (such as `12`). In other words, in a "show 12 rows, starting at index 1" scenario, we know that the table is rendering indices 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11 and 12.

To calculate that row, we therefore simply do: `(firstVisibleIdx + self.displayRows) - 1;`

Which in our example would be: `(1 + 12) - 1;` meaning `13 - 1`, meaning 12.

Which is correct! We're showing index 1 to 12 on-screen! :-)